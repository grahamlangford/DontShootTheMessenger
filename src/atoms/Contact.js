import React from 'react'
import PropTypes from 'prop-types'
import { Dimensions, StyleSheet, Text, View } from 'react-native'

const Contact = ({ onPress, item }) => {
  return (
    <Text style={styles.listItem} onPress={() => onPress(item)}>
      {item}
    </Text>
  )
}

const styles = StyleSheet.create({
  listItem: {
    height: 48,
    width: Dimensions.get('window').width,
    textAlignVertical: 'center',
    paddingLeft: 16
  }
})

Contact.propTypes = {
  onPress: PropTypes.func.isRequired,
  item: PropTypes.string.isRequired
}

export default Contact
