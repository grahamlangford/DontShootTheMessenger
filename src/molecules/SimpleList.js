import React from 'react'
import PropTypes from 'prop-types'
import { Dimensions, FlatList, StyleSheet, Text, View } from 'react-native'
import Contact from '../atoms/Contact'

const SimpleList = ({ itemClickHandler, items }) => {
  return (
    <View style={styles.body}>
      <FlatList
        data={items}
        renderItem={({ item }) => (
          <Contact onPress={itemClickHandler} item={item} />
        )}
        keyExtractor={(item, index) => index.toString()}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  body: {
    flex: 2,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center'
  },
  listItem: {
    height: 48,
    width: Dimensions.get('window').width,
    textAlignVertical: 'center',
    paddingLeft: 16
  }
})

SimpleList.propTypes = {
  itemClickHandler: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired
}

export default SimpleList
