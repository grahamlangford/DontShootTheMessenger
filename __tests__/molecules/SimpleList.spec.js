// __tests__/molecules/SimpleList.js
import { Text } from 'react-native'
import SimpleList from '../../src/molecules/SimpleList'
import Contact from '../../src/atoms/Contact'
import { React, shallow } from '../helpers/specHelper'

describe('SimpleList.js', () => {
  let wrapper
  const itemClickHandlerMock = jest.fn()
  const items = ['Picard', 'Riker', 'Data', 'Troi', 'La Forge', 'Worf']
  beforeEach(() => {
    jest.resetModules
    wrapper = shallow(
      <SimpleList itemClickHandler={itemClickHandlerMock} items={items} />
    )
  })

  it('renders', () => {
    expect(wrapper).toBeTruthy()
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
