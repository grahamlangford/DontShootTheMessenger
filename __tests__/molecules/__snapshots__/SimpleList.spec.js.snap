// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleList.js matches the snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <FlatList
    data={
      Array [
        "Picard",
        "Riker",
        "Data",
        "Troi",
        "La Forge",
        "Worf",
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Component>
`;
